frontend/src/metabase/dashboard/components/Dashboard.jsx

```diff
@@ -8,7 +8,7 @@ import DashboardHeader from "./DashboardHeader";
 import DashboardGrid from "./DashboardGrid";
 import LoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper";
 import { t } from "ttag";
-import Parameters from "metabase/parameters/components/Parameters";
+import Parameters from "metabase/parameters/components/Parameters/Parameters";
```

Comment:
See https://www.notion.so/56f6f59da3b647de89acb28ea9ebc424?v=cca33933b6244a7a94f256d1c110ce09&p=5e3272c9ec0e472493c7d0e1b9bd5cb3

Reply:
Why is this refactoring/cleanup lumped in the same PR as the bug fix? It makes any future troubleshooting/git bisect/git blame harder since the changes aren't separated.

Reply:
Thanks for your question @ariya 

This one is not completely trivial, if you allow me I am making a couple of videos to try to articulate the reasons: https://www.loom.com/share/250168729f444a04bb080b73c5a63c86

…tbc

Reply:
And a second one

https://www.loom.com/share/aecafb3ba23e41199b16bdc03b51803f

This is the branch with another version of the fix I mention in the video.

I really wish I could've been more succint on this.

Will be grateful to hear your thoughts on this.

Reply:
Sorry for the silly question/suggestion in advance, but after watching both videos I'm wondering is it possible to have fix in the first PR with the minimal changes (but let's keep the repro in it), and then immediately follow up with another PR that actually deals with the code clean up. That way we keep clean history, it's easy to revert and to review. Seems like a win-win to me.
Wdyt @ariya and @gusaiani .

Btw, thank you so much Gus for going an extra mile here and including two branches that offer alternative solutions 🙏

Reply:
Thanks @gusaiani for the explanations, it is really useful! To be fair, I think the fix is very decent.
I like the proposed solution from @nemanjaglumac. A minor variant of this, which I believe it's easier given the circumstance, is to create the first PR that does only file shuffling (as outlined in the referred Notion page), thereby still inhering the bug. The second PR, the follow-up, is the actual fix along with the repro.

Reply:
> to create the first PR that does only file shuffling (as outlined in the referred Notion page), thereby still inhering the bug. The second PR, the follow-up, is the actual fix along with the repro.

Thanks Nemanja and Ariya, I will do it like above.